server/controllers/request.js

```diff
@@ -0,0 +1,97 @@
+const asyncHandler = require("express-async-handler");
+const mongoose = require("mongoose");
+const Request = require("../models/Request");
+
+// @route GET /request
+// @desc gets all requests for loged-in dog sitter
+// @access Private
+exports.getRequests = asyncHandler(async (req, res, next) => {
+    const { id: sitterId } = req.user; 
+    
+    const requests = await Request.find({ sitterId })
+    .populate('userId')
+    .populate("sitterId")
+
+    if(!requests.length){
+        res.status(404)
+        throw new Error("No Request found for this user")
+    }
+
+    res.status(200).send(requests)
+});
+
+// @route POST /request
+// @desc dog-owner Create request for dog-sitters
+// @access Private
+exports.createRequest = asyncHandler(async (req, res, next) => {
+    const { id: dogOwnerId } = req.user; 
+    let { sitterId, start, end } = req.body
+
+    start = new Date(start)
+    end = new Date(end)
+
+    if(isNaN(start.getTime()) || isNaN(end.getTime())){
+        res.status(400)
+        throw new Error('Invalid start or end date')
+    }
+
+    // make sure the end date is always ahead of the start date
+    if( end.getTime() <= start.getTime()){
+        res.status(400)
+        throw new Error("End time must be ahead of the start time")
+    }
+
+    // make sure not to save the same request more than once
+    const checkRequest = await Request.findOne({ userId: dogOwnerId, sitterId, start, end })
```

Comment:
I don't think we will need this check, it entirely comes down to how the availability schema has been designed ( You can leave this is in for now though)